Source/BluetoothManager.swift

```diff
@@ -378,12 +344,19 @@ public class BluetoothManager {
     /// - Parameter peripheral: `Peripheral` which is monitored for disconnection.
     /// - Returns: Observable which emits next events when `Peripheral` instance was disconnected.
     /// It provides optional error which may contain more information about the cause of the disconnection
-    /// if it wasn't the `cancelConnection` call
+    /// if it wasn't the `cancelConnection` call. 
     public func monitorDisconnection(for peripheral: Peripheral) -> Observable<(Peripheral, DisconnectionReason?)> {
         let observable = delegateWrapper.didDisconnectPeripheral
           .filter { $0.0 == peripheral.peripheral }
           .map { (_, error) -> (Peripheral, DisconnectionReason?) in (peripheral, error) }
         return ensure(.poweredOn, observable: observable)
+                .catchError { error in
+                    if error is BluetoothError {
+                        return .concat(.just((peripheral, error)), .error(error))
```

Comment:
Do we really need to emit .error in the end?

Reply:
Don't we? I think sending error event when bluetooth state changed from .poweredOn is a must here.

Reply:
For single peripheral is probably makes sense.